### DISCUSSION TITLE
how to improve the memory of a windows 11 vm

### DISCUSSION THREAD

**Discussion by bluelottus** [utmapp/UTM#3653] [UTM Help (macOS)]
Hello I just install a windows 11 vm on UTM but I want more memory so how is it possible to improve it or adding an hard disk or an usb on the vm please.

**Configuration**
* UTM Version: 
* OS Version: 12.2.1
* Intel or Apple Silicon? Apple

**Upload VM**
(Optional) If possible, upload the `config.plist` inside the `.utm`. If you do not have this, you can upload the entire `.utm` but note this contains your personal data. Since Github has an attachment size limit, you may want to upload to another service such as Google Drive. Link it here.

**Comment by phirestalker** [comment]
After you right-click on the VM and choose edit, you go to system and what does your memory slider show? How much memory do you want to use?

Without knowing your version of UTM, I cannot help you with adding a drive.

**Comment by bluelottus** [comment]
hi thanks for reply. I actually use the last version of utm 
<img width="895" alt="Capture d’écran 2022-02-20 à 20 20 29" src="https://user-images.githubusercontent.com/100030740/154860301-4097b78a-c112-4e0b-aded-767e0186d32b.png">
but I can change the value of the memory and when i add a hard disk after 10 he is disconnected so I can't upload anything's on it because it is always disconnected during the upload and after a few seconds the vm crash 
<img width="448" alt="Capture d’écran 2022-02-20 à 20 25 12" src="https://user-images.githubusercontent.com/100030740/154860525-9501b05a-0648-41fc-8f0f-b1614910eaee.png">
I add my disk whit this and I already check the USB sharing option

**Comment by bluelottus** [comment]
sorry for my English I don't speak very well

**Comment by phirestalker** [comment]
Oh, I'm sorry, I know nothing about USB sharing. I guess you meant memory as in storage. I thought you were asking for help with increasing RAM AND adding storage.

**Comment by bluelottus** [comment]
no problème if you can explain me how to increase the ram of the vm it can be cool please

**Comment by phirestalker** [comment]
1. Right click on the VM name
2. Choose edit
3. In the window that opens, choose system on the left
4. Move the slider to the right to increase RAM for the VM (or type in the amount desired in MB)
5. Once you click save it should change the available RAM in your VM the next time you start it.
NOTE: if any of the VMs are 32 bit instead of 64 bit, it will not see RAM above about 4GB no matter what you set the slider to.

**Comment by bluelottus** [comment]
ok thank and how I can adding storage because I only have 64gb and I want like 100gb

**Comment by phirestalker** [comment]
In the same settings window, instead of clicking system click the "new drive" button at the bottom. Then just type in the size you want and click create. You will need to format it in the VM, but it should show up as another hardware drive.

**Comment by bluelottus** [comment]
ok i see but how I format it in the vm because it did'n appear

**Comment by bluelottus** [comment]
in the vm

**Comment by bluelottus** [comment]
can you show me me with some picture maybe

**Comment by phirestalker** [comment]
I'm afraid I don't use windows much anymore. You would be better off using a search engine for "how to format new drive in windows 10"

**Comment by bluelottus** [comment]
ok thanks you very much for your time an advises
